src/features/client/components/RestaurantInspectCard/index.jsx

```diff
@@ -7,12 +7,26 @@ import {
   useDisclosure,
   Button,
 } from '@chakra-ui/react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import { useState, useEffect } from 'react';
 import { addToCart } from '../../../../store/reducers/cartReducer';
 import RestaurantInspectModal from '../RestaurantInspectModal';
 
 function RestaurantInspectCard({ deal, cartOpened }) {
   const { isOpen, onOpen, onClose } = useDisclosure();
+  const currentItem = useSelector((store) =>
+    store.cart.list.find((item) => item.id === deal.id)
+  );
+  const [itemQuantity, setItemQuantity] = useState(
+    currentItem?.quantity ? currentItem.quantity : ''
+  );
```

Comment:
itemQuantity === currentItem.quantity. By creating a local state you're breaking single source of truth. I'd just deconstruct the currentItem and grab quantity from there, could rename to itemQuantity when decostructed for clearity.